### ISSUE TITLE
Custom PropertyNameResolver bug?

### ISSUE THREAD

**Issue by flodpanter** [JeremySkinner/FluentValidation#138]
I have an ASP.net MVC 5 application that uses Fluentvalidation. I use Ninject and my own validation factory to hook up Fluentvalidation with ASP.NET MVC. It works fine, however I am experiencing a problem when trying to define my own PropertyNameResolver to wrap all property names in single quotes. In other words, I want the error message to be " 'Name' is required" instead of "Name is required". This should be possible to do for all messages using the PropertyNameResolver. If I do try to do this then it looks like no client side rules are run. In short, the following line causes the problem:  
    
    ValidatorOptions.PropertyNameResolver = PropertyNameResolver;

In Ninjectwebcommon.cs

    FluentValidationModelValidatorProvider.Configure(provider => { provider.ValidatorFactory = new FluentValidationConfig(kernel);
			                                                                 provider.AddImplicitRequiredValidator = false;
    });

My validatorFactory:
    
    public class ValidatorFactory : ValidatorFactoryBase
    {
        private readonly IKernel _kernel;

        
        public ValidatorFactory(IKernel kernel)
        {
            _kernel = kernel;

            ValidatorOptions.ResourceProviderType = typeof(ErrorMessages);
            //the below line causes the client side validation rules not to run!
            ValidatorOptions.PropertyNameResolver = PropertyNameResolver;
            ValidatorOptions.CascadeMode = CascadeMode.Continue;

            AddBindings();
        }

        public string PropertyNameResolver(Type type, MemberInfo memberInfo, LambdaExpression arg3)
        {
            if (memberInfo != null)
            {
                return "xyz";
            }
            return null;
        }

        public void AddBindings()
        {
            _kernel.Bind<IValidator<MyViewModel>>().To<MyValidator>();
        }

        public override IValidator CreateInstance(Type validatorType)
        {
            if (validatorType == null)
            {
                return null;
            }
            var validator= (IValidator)_kernel.TryGet(validatorType);

            return validator;
        }
    }

**Comment by flodpanter** [comment]
The way I define a Rule:

     RuleFor(x => x.Attention).
                Length(0, 60).
                WithName(Resources.Creditor.Creditor_label_Attention);

This gives me an error message where the value of Resources.Creditor.Creditor_label_Attention is merged into the message as expected. I have overridden the standard error message of the Length-rule and this works fine as well. 

    {PropertyName} must be shorter than {MaxLength}

So to sum up, the error message seems provided by the localization methods of FluentValidation. Am I missing something?

**Comment by JeremySkinner** [comment]
Ah I see. Can you not just put the single quotes inside your custom resources? The default localization messages do include the quotes, so it would make sense if you also put them inside yours for consistency.

That being said, if you really don't want to do this, then you can use the resolver to do it, but should use the DisplayNameResolver rather than the PropertyNameResolver.

**Comment by flodpanter** [comment]
I cannot get it to work with the DisplayNameResolver. This alters the Property name of the model error, hence making it impossible for me to highlight the related html control. Furthermore this does not seem to alter the actual validation error message. I will go with putting the quotes in the resource file in stead. Cheers
